### ISSUE TITLE
Adds '#/definitions' to a '$ref' field when it is a URL too

### ISSUE THREAD

**Issue by saravanak** [fotinakis/swagger-blocks#61]
I have a block like this:

```
      response 200 do
        key :description, 'List of tasks filtered, sorted and paged accordingly'
        schema do
          key :type ,:array
          key :description , "test"
          items do
           key :'$ref', "http://example/api/v1/schema.json#/definitions/model"
           key :description , "test"
         end
        end
      end
```

But the resolved JSON is generated as:
```
key :'$ref', "#/defintions/http://example/api/v1/schema.json#/definitions/model"
```

#29 is related to this, and I assume that we need to check for existence of `[http/https] ` before prepending the definitions part

**Comment by fotinakis** [comment]
Digging into this a bit — there is no example for a URI in a `$ref` field in the Swagger 2.0 spec. It does say "The Reference Object is a [JSON Reference](http://tools.ietf.org/html/draft-pbryan-zyp-json-ref-02) that uses a [JSON Pointer](http://tools.ietf.org/html/rfc6901) as its value." and if you read the JSON Pointer docs, it looks like it can only be a "fragment identifier" and not a full URI. 

So, before you go further down this road or before we support it, are you sure it's valid and/or supported by the Swagger spec? :) If it is, or if something like Swagger UI implicitly supports it, I'm fine with updating swagger-blocks to not prepend `#/definitions` in this case (it's an easy change).

**Comment by saravanak** [comment]
The [JSON Reference - Section 3](http://tools.ietf.org/html/draft-pbryan-zyp-json-ref-02#section-3)  has an example with a full URI. The next section, `Resolution` also talks about how to resolve the URI and establish the relative context from it.  So IMO, this should be fixed to align with the spec. 

The Swagger UI (or rather the swagger-client) definitely supports this - it resolved the URL properly when  I fed the JSON representation of the ` "http://example/api/v1/schema.json#/definitions/model"`.

**Comment by fotinakis** [comment]
Sounds good — they should clarify in the actual Swagger spec what is supposed to be supported, since apparently the JSON Pointer spec does not allow URIs but the JSON Reference spec does.

Fixed in https://github.com/fotinakis/swagger-blocks/commit/072dd59139a71d8e2b80496721876da9b028263d and released as gem v1.3.4.
